srcs/hooks/key_hook.c

```diff
@@ -1,29 +1,82 @@
 #include "cub.h"
 
-int	key_hook(int keycode, t_cub *cub)
+int	key_up(int keycode, t_cub *cub)
 {
-	double	edge;
-	double	move_speed;
-	double	rot_speed;
+	if (keycode == W_KEY)
+		cub->keys.w = 0;
+	if (keycode == A_KEY)
+		cub->keys.a = 0;
+	if (keycode == S_KEY)
+		cub->keys.s = 0;
+	if (keycode == D_KEY)
+		cub->keys.d = 0;
+	if (keycode == UP_KEY)
+		cub->keys.up = 0;
+	if (keycode == DOWN_KEY)
+		cub->keys.down = 0;
+	if (keycode == LEFT_KEY)
+		cub->keys.left = 0;
+	if (keycode == RIGHT_KEY)
+		cub->keys.right = 0;
+	return (0);
+}
 
-	edge = 1.1;
-	move_speed = 0.6;
-	rot_speed = 0.1;
-	if (keycode == ESC_KEY)
-		graceful_exit(cub);
-	else if (keycode == W_KEY || keycode == UP_KEY)
-		move_forward(cub, edge, move_speed);
-	else if (keycode == S_KEY || keycode == DOWN_KEY)
-		move_backward(cub, edge, move_speed);
-	else if (keycode == LEFT_KEY)
-		rotate_left(cub, rot_speed);
-	else if (keycode == RIGHT_KEY)
-		rotate_right(cub, rot_speed);
-	else if (keycode == A_KEY)
-		move_left(cub, edge, move_speed);
-	else if (keycode == D_KEY)
-		move_right(cub, edge, move_speed);
-	raycast(cub);
-	mlx_put_image_to_window(cub->mlx, cub->window, cub->img, 0, 0);
+int	key_down(int keycode, t_cub *cub)
+{
+	if (keycode == W_KEY)
+		cub->keys.w = 1;
+	if (keycode == A_KEY)
+		cub->keys.a = 1;
+	if (keycode == S_KEY)
+		cub->keys.s = 1;
+	if (keycode == D_KEY)
+		cub->keys.d = 1;
+	if (keycode == UP_KEY)
+		cub->keys.up = 1;
+	if (keycode == DOWN_KEY)
+		cub->keys.down = 1;
+	if (keycode == LEFT_KEY)
+		cub->keys.left = 1;
+	if (keycode == RIGHT_KEY)
+		cub->keys.right = 1;
 	return (0);
 }
+
+/*
+	UPDATE_DISPLAY:
+	-	Inside of mlx_loop_hook, so happening when NO event happens
+	-	Checks the key value in the relevant struct set by key_down and key_up
+	-	If values in the struct are non-zero, apply movement to coords
+	
+	-	VAR: Edge detection = 1.1]
+	-	VAR: Movement Speed = 0.05 squares/frame]
+	-	VAR: Rotation Speed = 0.02 radians]
+
+	-	Above values were set by experimentation. We can also change the
+		static int frame value in the if statement
+*/
+int update_display(t_cub *cub)
```

Comment:
not sure I'd put this in this file (I mean, it's not a key hook). Not sure where I'd put it though

Reply:
Could be yes, I think that would do the trick